### PR TITLE
#1417 - Fix profile completion field data not updating

### PR DESCRIPTION
**Pull request by chetansatasiya**
This reverts commit 0b08d1365e0edebfb2ac3c5ff10a04ae0507c409.

### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #ISSUE_NUMBER .

### How to test the changes in this Pull Request:

1.
2.
3.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Comment by mohit-buddyboss** [comment]
@dipskakadiya 
Kindly review this PR and Related theme PR buddyboss/buddyboss-theme/pull/1137.

**Review by dipskakadiya — DISMISSED** [review]
@chetansatasiya Approved. Merge this PR into `dev` and assign to @mohit-buddyboss

**Review by dipskakadiya — DISMISSED** [review]
@chetansatasiya Approved

**Comment by chetansatasiya** [comment]
@ankitguptaindia This is ready for testing merged changes to dev.

**Comment by mohit-buddyboss** [comment]
@chetansatasiya 
FIXED> Tested this again on dev branch after latest merge. Also tested linked PR : buddyboss/buddyboss-theme/pull/1137 related issue.
Tested this with and without Redis enable on server and by activating and deactivating Redis object cache Plugin as well as enabling and disabling website object cache. Also, by adding and deleting Profile fields.

Videos:
https://www.loom.com/share/c43a6df0193245e49778fe2b1cd9d36b - without redis enable
https://www.loom.com/share/43ad7617101f4189b1b2f6946af72b36 - with Redis enable and Plugin active
https://www.loom.com/share/115acbe66ce145edad4d28d752bd329c - with cache active

@mehulkaklotar - Related issue is Fixed and PR is ready to merge.

**Review by mohit-buddyboss — APPROVED** [review]
Related issue is resolve now and PR is ready to merge.

@mehulkaklotar

File: src/bp-core/bp-core-functions.php [hunk 1]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+
+	if ( $total_count > 0 ) {
+		$response['total_fields'] = $total_count;
+	}
+
+	if ( $total_completed_count > 0 ) {
+		$response['completed_fields'] = $total_completed_count;
+	}
+
+	return apply_filters( 'bp_xprofile_get_selected_options_user_progress', $response, $profile_groups, $profile_photo_type, $get_user_data );
```
Comment (mehulkaklotar) [inline src/bp-core/bp-core-functions.php:5176]: @chetansatasiya please add filter doc here.

File: src/bp-core/bp-core-functions.php [hunk 2]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+			}
+		}
+	}
 
-	$keys = $wpdb->get_col( $sql );
+	if ( ! empty( $profile_groups ) ) {
+		foreach ( $profile_groups as $group ) {
+			if ( isset( $get_user_data['groups'][$group] ) ) {
+				$response['groups'][$group] = $get_user_data['groups'][$group];
+				$total_count = $total_count + (int) $get_user_data['groups'][$group]['group_total_fields'];
```
Comment (mehulkaklotar) [inline src/bp-core/bp-core-functions.php:5159]: ```phpcs: WordPress.Arrays.ArrayKeySpacingRestrictions.NoSpacesAroundArrayKeys: Array keys must be surrounded by spaces unless they contain a string or an integer.```

File: src/bp-core/bp-core-functions.php [hunk 3]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+				if ( isset( $get_user_data['groups'][$group]['group_completed_fields'] ) && (int) $get_user_data['groups'][$group]['group_completed_fields'] > 0 ) {
```
Comment (mehulkaklotar) [inline src/bp-core/bp-core-functions.php:5160]: ```phpcs: WordPress.Arrays.ArrayKeySpacingRestrictions.NoSpacesAroundArrayKeys: Array keys must be surrounded by spaces unless they contain a string or an integer.```

File: src/bp-xprofile/bp-xprofile-functions.php [hunk 4]
```diff
@@ -2002,43 +2002,26 @@ function bp_xprofile_get_profile_completion_transient_name( $key, $widget_id ) {
...
-	} else {
-
-		// Get logged in user Progress.
-		$user_progress_arr = bp_xprofile_get_user_progress( $profile_groups, $profile_phototype );
-
-		// Do not proceed if no fields found based on settings.
-		if( isset( $user_progress_arr['total_fields'] ) && $user_progress_arr['total_fields'] <= 0  ){
-			return $user_progress;
-		}
+	$user_progress_arr = bp_xprofile_get_selected_options_user_progress( $settings );
```
Comment (mehulkaklotar) [inline src/bp-xprofile/bp-xprofile-functions.php:2015]: @chetansatasiya change variable name here
```suggestion
	$user_progress_options = bp_xprofile_get_selected_options_user_progress( $settings );
```

File: src/bp-xprofile/bp-xprofile-functions.php [hunk 5]
```diff
@@ -2002,43 +2002,26 @@ function bp_xprofile_get_profile_completion_transient_name( $key, $widget_id ) {
...
 
-		// Format User Progress array to pass on to the template.
-		$user_progress = bp_xprofile_get_user_progress_formatted( $user_progress_arr );
-
-		// set Transient here with 3hours expiration.
-		set_transient( $pc_transient_name, $user_progress, HOUR_IN_SECONDS * 3 );
+	// Do not proceed if no fields found based on settings.
+	if( isset( $user_progress_arr['total_fields'] ) && $user_progress_arr['total_fields'] <= 0  ){
```
Comment (mehulkaklotar) [inline src/bp-xprofile/bp-xprofile-functions.php:2018]:
```suggestion
	if( isset( $user_progress_options['total_fields'] ) && $user_progress_options['total_fields'] <= 0  ){
```

File: src/bp-xprofile/bp-xprofile-functions.php [hunk 6]
```diff
@@ -2002,43 +2002,26 @@ function bp_xprofile_get_profile_completion_transient_name( $key, $widget_id ) {
...
+		return $user_progress;
 	}
 
+	// Format User Progress array to pass on to the template.
+	$user_progress = bp_xprofile_get_user_progress_formatted( $user_progress_arr );
```
Comment (mehulkaklotar) [inline src/bp-xprofile/bp-xprofile-functions.php:2023]:
```suggestion
	$user_progress = bp_xprofile_get_user_progress_formatted( $user_progress_options );
```

File: src/bp-xprofile/classes/class-bp-xprofile-profile-completion-widget.php [hunk 7]
```diff
@@ -49,12 +49,14 @@ function widget( $args, $instance ) {
...
-		$profile_hide_widget_selected  = ! empty( $instance['profile_hide_widget'] ) ? $instance['profile_hide_widget'] : array();
-		$user_progress                = bp_xprofile_get_user_progress_data( $profile_groups_selected, $profile_phototype_selected, $this->widget_id );
+		$profile_groups_selected         = $instance['profile_groups_enabled'];
+		$this->widget_id                = $args['widget_id'];
+		$profile_phototype_selected      = ! empty( $instance['profile_photos_enabled'] ) ? $instance['profile_photos_enabled'] : array();
+		$profile_hide_widget_selected    = ! empty( $instance['profile_hide_widget'] ) ? $instance['profile_hide_widget'] : array();
+		$settings                       = array();
+		$settings['profile_groups']      = $profile_groups_selected;
+		$settings['profile_photo_type']  = $profile_phototype_selected;
+		$user_progress                  = bp_xprofile_get_user_profile_progress_data( $settings );
```
Comment (mehulkaklotar) [inline src/bp-xprofile/classes/class-bp-xprofile-profile-completion-widget.php:59]: check phpcs here

**Review by mehulkaklotar — CHANGES_REQUESTED** [review]
@chetansatasiya please follow PHPCS here and fix all the lint errors.

**Comment by ankitguptaindia** [comment]
@chetansatasiya Can you update after @mehulkaklotar suggestion so we can add in upcoming release?
As this is already tested by QA.


@mohit-buddyboss keep eye on this if we can add in next release after update by Chetan.

File: src/bp-core/bp-core-functions.php [hunk 8]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+ */
+function bp_core_profile_completion_steps_options() {
+
+	/* Profile Groups and Profile Cover Photo VARS. */
+	$options                              = array();
+	$options['profile_groups']            = bp_xprofile_get_groups();
+	$options['is_profile_photo_disabled'] = bp_disable_avatar_uploads();
+	$options['is_cover_photo_disabled']   = bp_disable_cover_image_uploads();
+
+	return apply_filters( 'bp_core_profile_completion_steps_options', $options );
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php:5088]: done

File: src/bp-xprofile/bp-xprofile-functions.php (hunk 6 above)
Comment (chetansatasiya) [reply src/bp-xprofile/bp-xprofile-functions.php]: done

File: src/bp-xprofile/bp-xprofile-functions.php (hunk 5 above)
Comment (chetansatasiya) [reply src/bp-xprofile/bp-xprofile-functions.php]: done

File: src/bp-xprofile/bp-xprofile-functions.php (hunk 4 above)
Comment (chetansatasiya) [reply src/bp-xprofile/bp-xprofile-functions.php]: done

File: src/bp-xprofile/classes/class-bp-xprofile-profile-completion-widget.php (hunk 7 above)
Comment (chetansatasiya) [reply src/bp-xprofile/classes/class-bp-xprofile-profile-completion-widget.php]: done

File: src/bp-core/bp-core-functions.php (hunk 1 above)
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php:5195]: done

File: src/bp-core/bp-core-functions.php (hunk 3 above)
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php (hunk 2 above)
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php [hunk 9]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+				}
+			}
+		}
+	}
 
-	$keys = $wpdb->get_col( $sql );
+	if ( ! empty( $profile_groups ) ) {
+		foreach ( $profile_groups as $group ) {
+			if ( isset( $get_user_data['groups'][$group] ) ) {
+				$response['groups'][$group] = $get_user_data['groups'][$group];
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php [hunk 10]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+					$total_completed_count = $total_completed_count + 1;
+				}
+			}
+		}
+	}
 
-	$keys = $wpdb->get_col( $sql );
+	if ( ! empty( $profile_groups ) ) {
+		foreach ( $profile_groups as $group ) {
+			if ( isset( $get_user_data['groups'][$group] ) ) {
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php [hunk 11]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+				$response['photo_type']['profile_photo'] = $get_user_data['photo_type']['profile_photo'];
+				$total_count = $total_count + 1;
+				if ( isset( $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) &&  1 === (int) $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) {
+					$total_completed_count = $total_completed_count + 1;
+				}
+			} elseif ( 'cover_photo' === $option && isset( $get_user_data['photo_type'] ) && isset( $get_user_data['photo_type']['cover_photo'] ) ) {
+				$response['photo_type']['cover_photo'] = $get_user_data['photo_type']['cover_photo'];
+				$total_count = $total_count + 1;
+				if ( isset( $get_user_data['photo_type']['cover_photo']['is_uploaded'] ) &&  1 === (int) $get_user_data['photo_type']['cover_photo']['is_uploaded'] ) {
+					$total_completed_count = $total_completed_count + 1;
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php [hunk 12]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+			if ( 'profile_photo' === $option && isset( $get_user_data['photo_type'] ) && isset( $get_user_data['photo_type']['profile_photo'] ) ) {
+				$response['photo_type']['profile_photo'] = $get_user_data['photo_type']['profile_photo'];
+				$total_count = $total_count + 1;
+				if ( isset( $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) &&  1 === (int) $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) {
+					$total_completed_count = $total_completed_count + 1;
+				}
+			} elseif ( 'cover_photo' === $option && isset( $get_user_data['photo_type'] ) && isset( $get_user_data['photo_type']['cover_photo'] ) ) {
+				$response['photo_type']['cover_photo'] = $get_user_data['photo_type']['cover_photo'];
+				$total_count = $total_count + 1;
+				if ( isset( $get_user_data['photo_type']['cover_photo']['is_uploaded'] ) &&  1 === (int) $get_user_data['photo_type']['cover_photo']['is_uploaded'] ) {
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php [hunk 13]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+		foreach ( $profile_photo_type as $option ) {
+			if ( 'profile_photo' === $option && isset( $get_user_data['photo_type'] ) && isset( $get_user_data['photo_type']['profile_photo'] ) ) {
+				$response['photo_type']['profile_photo'] = $get_user_data['photo_type']['profile_photo'];
+				$total_count = $total_count + 1;
+				if ( isset( $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) &&  1 === (int) $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) {
+					$total_completed_count = $total_completed_count + 1;
+				}
+			} elseif ( 'cover_photo' === $option && isset( $get_user_data['photo_type'] ) && isset( $get_user_data['photo_type']['cover_photo'] ) ) {
+				$response['photo_type']['cover_photo'] = $get_user_data['photo_type']['cover_photo'];
+				$total_count = $total_count + 1;
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php [hunk 14]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+	$total_count                  = 0;
+	$total_completed_count        = 0;
+
+	if ( ! empty( $profile_photo_type ) ) {
+		foreach ( $profile_photo_type as $option ) {
+			if ( 'profile_photo' === $option && isset( $get_user_data['photo_type'] ) && isset( $get_user_data['photo_type']['profile_photo'] ) ) {
+				$response['photo_type']['profile_photo'] = $get_user_data['photo_type']['profile_photo'];
+				$total_count = $total_count + 1;
+				if ( isset( $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) &&  1 === (int) $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) {
+					$total_completed_count = $total_completed_count + 1;
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php [hunk 15]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+	$response['completed_fields'] = 0;
+	$total_count                  = 0;
+	$total_completed_count        = 0;
+
+	if ( ! empty( $profile_photo_type ) ) {
+		foreach ( $profile_photo_type as $option ) {
+			if ( 'profile_photo' === $option && isset( $get_user_data['photo_type'] ) && isset( $get_user_data['photo_type']['profile_photo'] ) ) {
+				$response['photo_type']['profile_photo'] = $get_user_data['photo_type']['profile_photo'];
+				$total_count = $total_count + 1;
+				if ( isset( $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) &&  1 === (int) $get_user_data['photo_type']['profile_photo']['is_uploaded'] ) {
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php [hunk 16]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
+	$response['total_fields']     = 0;
+	$response['completed_fields'] = 0;
+	$total_count                  = 0;
+	$total_completed_count        = 0;
+
+	if ( ! empty( $profile_photo_type ) ) {
+		foreach ( $profile_photo_type as $option ) {
+			if ( 'profile_photo' === $option && isset( $get_user_data['photo_type'] ) && isset( $get_user_data['photo_type']['profile_photo'] ) ) {
+				$response['photo_type']['profile_photo'] = $get_user_data['photo_type']['profile_photo'];
+				$total_count = $total_count + 1;
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

File: src/bp-core/bp-core-functions.php [hunk 17]
```diff
@@ -5066,36 +5066,130 @@ function bp_core_upload_max_size() {
...
-function bp_core_delete_transient_query( $transient_name_prefix ) {
-	global $wpdb;
-	$sql = $wpdb->prepare(
-			"SELECT `option_name` FROM {$wpdb->options} WHERE option_name LIKE '%s' ",
-			$transient_name_prefix
-	);
+function bp_core_xprofile_update_profile_completion_user_progress() {
+
+	$user_id            = get_current_user_id();
+	$steps_options     = bp_core_profile_completion_steps_options();
```
[the remark this answers is not in the document]
Comment (chetansatasiya) [reply src/bp-core/bp-core-functions.php]: done

**Comment by chetansatasiya** [comment]
@mehulkaklotar All of requested changes done. It's ready for release.

**Comment by mohit-buddyboss** [comment]
@chetansatasiya 
After latest changes tested again it is working as before and as expected.

@mehulkaklotar @ankitguptaindia

**Review by mehulkaklotar — APPROVED** [review]
approved! 🚀

**Pull request merged**